scalanet/src/io/iohk/scalanet/peergroup/kademlia/Xor.scala

```diff
@@ -0,0 +1,23 @@
+package io.iohk.scalanet.peergroup.kademlia
+
+import scodec.bits.BitVector
+
+object Xor {
+
+  def d(a: BitVector, b: BitVector): Int = {
+    assert(a.length == b.length)
+    a.length.toInt - leadingZeros(a.xor(b))
```

Comment:
as discussed before, this is not Kademlia's paper metric. It does not satisfy unidirectionality.